### PR TITLE
#CU-861xvx Components/richeditor

### PR DESCRIPTION
**Pull request by binhtt13122000**
Do editor and update feature border for box
image
#CU-861xvx

### CODE REVIEW

**Comment by binh1298** [comment]
Task linked: [CU-861xvx Research/Implement richtext editor using a WYSIWYG editor](https://app.clickup.com/t/861xvx)

File: package.json [hunk 1]
```diff
@@ -10,13 +10,19 @@
...
         "@types/react": "^17.0.0",
         "@types/react-dom": "^17.0.0",
+        "@types/react-draft-wysiwyg": "^1.13.2",
+        "draft-js": "^0.11.7",
+        "draftjs-to-html": "^0.9.1",
+        "draftjs-to-markdown": "^0.6.0",
         "jwt-decode": "^3.1.2",
         "react": "^17.0.2",
         "react-dom": "^17.0.2",
+        "react-draft-wysiwyg": "^1.14.7",
```
Comment (binh1298) [inline package.json:25]: We need all of these dependencies to make this work?

File: src/setupTests.ts [hunk 2]
```diff
@@ -3,3 +3,6 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import "@testing-library/jest-dom";
+
+const noop = () => {};
+Object.defineProperty(window, "scrollTo", { value: noop, writable: true });
```
Comment (binh1298) [inline src/setupTests.ts:8]: Why do we need this?

File: src/components/Boxs/BoxBase/index.tsx [hunk 3]
```diff
@@ -96,13 +96,15 @@ export interface BoxBaseProps extends MaterialBoxProps {
...
     overrideColor?: "primary" | "secondary" | "info" | "error" | "warning" | "success";
+    border?: boolean;
 }
 
 const BoxBase: React.FC<BoxBaseProps> = ({
     bgcolor,
     colorProps = "main",
     color,
     overrideColor,
+    border,
```
Comment (binh1298) [inline src/components/Boxs/BoxBase/index.tsx:107]:
```suggestion
    shouldHasBorder,
```
boolean should start with is or should

File: src/components/RichEditor/RichEditorBase/RichEditorBase.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+} as Meta;
+
+const Template: Story<RichEditorBaseProps> = (args) => {
+    const [value, setValue] = useState<string>("");
+    const getEditorValue = (content: string) => {
+        setValue(content);
+    };
+    const { getValue, ...rest } = args;
+    return (
+        <BoxBase width={1} display="flex">
```
Comment (binh1298) [inline src/components/RichEditor/RichEditorBase/RichEditorBase.stories.tsx:22]: What does 1 mean here?

File: src/components/Boxs/BoxBase/index.tsx (hunk 3 above)
Comment (binhtt13122000) [reply src/components/Boxs/BoxBase/index.tsx:107]: Ok, it's my mistake!

File: src/components/RichEditor/RichEditorBase/RichEditorBase.stories.tsx (hunk 4 above)
Comment (binhtt13122000) [reply src/components/RichEditor/RichEditorBase/RichEditorBase.stories.tsx:22]: w={1} is same width="100%".
I had demos of width on stories about size of boxbase and wrote description in width field.
image

File: package.json (hunk 1 above)
Comment (binhtt13122000) [reply package.json:25]: I had removed unused dependencies.

Comment (binhtt13122000) [reply package.json:25]: image

File: src/setupTests.ts (hunk 2 above)
Comment (binhtt13122000) [reply src/setupTests.ts:8]: When I using yarn test to test my unit test of richeditor, the testcases passed 100% but some errors happened.
image
When I read documents, I realiezed I need to mock window.scrollTo in each test case need to use ref.
image
So i set here to effect all test case

File: src/components/RichEditor/RichEditorBase/RichEditorBase.stories.tsx (hunk 4 above)
Comment (binh1298) [reply src/components/RichEditor/RichEditorBase/RichEditorBase.stories.tsx:22]: Cool

File: package.json [hunk 5]
```diff
@@ -14,9 +14,12 @@
         "@types/node": "^12.0.0",
         "@types/react": "^17.0.0",
         "@types/react-dom": "^17.0.0",
+        "@types/react-draft-wysiwyg": "^1.13.2",
```
Comment (binh1298) [inline package.json:17]: Please install this as a devDependency

File: src/components/RichEditor/RichEditorBase/index.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+const RichEditorBase = (props: RichEditorBaseProps) => {
+    const editorRef = useRef<HTMLInputElement>();
+
+    useEffect(() => {
+        if (editorRef.current && props.focus) {
+            editorRef.current.focus();
+        }
+    }, [props.focus]);
```
Comment (binh1298) [inline src/components/RichEditor/RichEditorBase/index.tsx:25]: Why do we need this part?

**Comment by binh1298** [comment]
Can we convert the content to HTML format?

File: src/components/RichEditor/RichEditorBase/index.tsx (hunk 6 above)
Comment (binhtt13122000) [reply src/components/RichEditor/RichEditorBase/index.tsx:25]: I think It need to do some features like focus and show error when invalid input,...

File: package.json (hunk 5 above)
Comment (binhtt13122000) [reply package.json]: image
I done it

**Comment by binhtt13122000** [comment]
> Can we convert the content to HTML format?

I think if we save with html, we can be attacked by XSS, so I do a component to display content of base editor base on the editor with set editor is read only and hide the toolbar.

**Review by binh1298 — APPROVED** [review]

**Comment by binh1298** [comment]
Very good

**Pull request merged**